services-operator/test/test_beamsqlstatementsetoperator.py

```diff
@@ -0,0 +1,308 @@
+                },
+                "spec": {
+                    "sqlstatements": ["select;"],
+                    "tables": ["table"]
+                },
+                "status": {
+                    "state": "CANCELED",
+                    "job_id": "job_id"
+                }
+        }
+        patch = Bunch()
+        patch.status = {}
+        
+        beamsqltables = {}
+        target.delete(body, body["spec"], patch, Logger())
+        assert patch.status == {}
+
+    @patch('kopf.info', kopf_info)
+    @patch('beamsqlstatementsetoperator.refresh_state', update_status_nochange)
+    def test_canceling_delete(self):
+        body = {
+                "metadata": {
+                    "name": "name",
+                    "namespace": "namespace"
+                },
+                "spec": {
+                    "sqlstatements": ["select;"],
+                    "tables": ["table"]
+                },
+                "status": {
+                    "state": "CANCELING",
+                    "job_id": "job_id"
+                }
+        }
+        patch = Bunch()
+        patch.status = {"state": ""}
+
+        beamsqltables = {}
+        try:
+            target.delete(body, body["spec"], patch, Logger())
+        except Exception as err:
+            assert type(err) == kopf.TemporaryError
+            assert str(err).startswith("Canceling,")
+
+    @patch('kopf.info', kopf_info)
+    @patch('beamsqlstatementsetoperator.refresh_state', update_status_change)
+    def test_canceling_delete(self):
+        body = {
+                "metadata": {
+                    "name": "name",
+                    "namespace": "namespace"
+                },
+                "spec": {
+                    "sqlstatements": ["select;"],
+                    "tables": ["table"]
+                },
+                "status": {
+                    "state": "CANCELING",
+                    "job_id": "job_id"
+                }
+        }
+        patch = Bunch()
+        patch.status = {"state": ""}
+
+        beamsqltables = {}
+
+        target.delete(body, body["spec"], patch, Logger())
+    @patch('flink_util.cancel_job', cancel_job)
+    @patch('beamsqlstatementsetoperator.refresh_state', update_status_nochange)
+    @patch('kopf.info', kopf_info)
+    def test_canceling_delete_flink_ok(self):
+        body = {
+                "metadata": {
+                    "name": "name",
+                    "namespace": "namespace"
+                },
+                "spec": {
+                    "sqlstatements": ["select;"],
+                    "tables": ["table"]
+                },
+                "status": {
+                    "state": "RUNNING",
+                    "job_id": "job_id"
+                }
+        }
+        patch = Bunch()
+        patch.status = {"state": ""}
+
+        beamsqltables = {}
+        try:
+            target.delete(body, body["spec"], patch, Logger())
+        except Exception as err:
```

Comment:
Is this required to happen, if so, better make sure with (see the accepted answer [here](https://stackoverflow.com/questions/129507/how-do-you-test-that-a-python-function-throws-an-exception))